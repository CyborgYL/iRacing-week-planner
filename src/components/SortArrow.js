// @flow

import * as React from 'react';

import type { sort as sortType } from '../reducers/settings';

import styles from '../styles/main.module.scss';

type Props = {
  sort: sortType,
};

export default function SortArrow({ sort }: Props): React.Node {
  if (sort.order === 'desc') {
    return <span className={`${styles.glyphicon} ${styles['glyphicon-triangle-bottom']}`} />;
  }

  return <span className={`${styles.glyphicon} ${styles['glyphicon-triangle-top']}`} />;
}
