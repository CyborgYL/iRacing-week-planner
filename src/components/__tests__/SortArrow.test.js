import React from 'react';
import { shallow } from 'enzyme';
import { describe, test } from '@jest/globals';
import SortArrow from '../SortArrow';

describe('components/SortArrow', () => {
  test('renders down correctly', () => {
    const component = shallow(<SortArrow sort={{ order: 'desc', key: 'licence' }} />);

    expect(component.find('.glyphicon-triangle-bottom').length).toBe(1);
    expect(component).toMatchSnapshot();
  });

  test('renders up correctly', () => {
    const component = shallow(<SortArrow sort={{ order: 'asc', key: 'licence' }} />);

    expect(component.find('.glyphicon-triangle-top').length).toBe(1);
    expect(component).toMatchSnapshot();
  });
});
