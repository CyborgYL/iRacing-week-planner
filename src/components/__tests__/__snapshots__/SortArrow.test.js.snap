// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/SortArrow renders down correctly 1`] = `
<span
  className="glyphicon glyphicon-triangle-bottom"
/>
`;

exports[`components/SortArrow renders up correctly 1`] = `
<span
  className="glyphicon glyphicon-triangle-top"
/>
`;
